### PR TITLE
SLING-8720 : HTTP Status 200 returned when JSP Engine fails due to Js…

### PR DESCRIPTION
**Pull request by ashokpanghal**
…pFactoryHandler being null

### CODE REVIEW

**Comment by ashokpanghal** [comment]
@cziegeler , thanks for the review. looks like sonar code smell of this PR is failing as it expects a non-generic exception to be thrown. can we merge it as is or need to change something here.

File: src/main/java/org/apache/sling/scripting/jsp/JspScriptEngineFactory.java [hunk 1]
```diff
@@ -323,12 +324,12 @@ private void callJsp(final Bindings bindings,
...
         final SlingIOProvider io = this.ioProvider;
         final JspFactoryHandler jspfh = this.jspFactoryHandler;
-
         // abort if JSP Support is shut down concurrently (SLING-2704)
         if (io == null || jspfh == null) {
             logger.warn("callJsp: JSP Script Engine seems to be shut down concurrently; not calling {}",
                     scriptHelper.getScript().getScriptResource().getPath());
-            return;
+            throw new Exception("callJsp: JSP Script Engine seems to be shut down concurrently; not calling {"+
+                    scriptHelper.getScript().getScriptResource().getPath()+"}");
```
Comment (rombert) [inline src/main/java/org/apache/sling/scripting/jsp/JspScriptEngineFactory.java]: You don't need { and } here anymore, the placeholder were replaced by the logging engine in the previous form.

File: src/main/java/org/apache/sling/scripting/jsp/JspScriptEngineFactory.java [hunk 2]
```diff
@@ -272,7 +272,8 @@ private void callErrorPageJsp(final Bindings bindings,
         if (io == null || jspfh == null) {
             logger.warn("callJsp: JSP Script Engine seems to be shut down concurrently; not calling {}",
                     scriptHelper.getScript().getScriptResource().getPath());
-            return;
+            throw new Exception("callJsp: JSP Script Engine seems to be shut down concurrently; not calling {"+
```
Comment (rombert) [inline src/main/java/org/apache/sling/scripting/jsp/JspScriptEngineFactory.java]: You don't need { and } here anymore, the placeholder were replaced by the logging engine in the previous form.

Comment (ashokpanghal) [reply src/main/java/org/apache/sling/scripting/jsp/JspScriptEngineFactory.java]: @rombert  true , I missed that . will remove . Thanks

**Comment by ashokpanghal** [comment]
@rombert , kindly check why PR merge job is broken. any way to re-trigger this. Thank you.

**Comment by rombert** [comment]
@ashokpanghal - there was a (hopefully transient) problem with the infrastructure. I have retriggered the build.

**Pull request merged**